### DISCUSSION TITLE
ValidateWindow called way too much?

### DISCUSSION THREAD

**Discussion by derwin12** [xLightsSequencer/xLights#4752] [Q&A]
While working on an issue I noted that ValidateWindow is called repeatedly. During startup and during a sequence load.
I added the wxLogMessage to EffectPanelUtils.h. The sequence had only 1 effect, no picture,pinwheel,ripple etc. 
Putting this on here to investigate further or if someone already knows the reason.

```
    void OnValidateWindow(wxCommandEvent& event) {
        wxLogMessage(this->GetName());
        ValidateWindow();
    }
```

> 14:26:23: ID_PANEL_PICTURES
> 14:26:23: ID_PANEL_PICTURES
> 14:26:23: ID_PANEL_PINWHEEL
> 14:26:23: ID_PANEL_PINWHEEL
> 14:26:23: ID_PANEL_PINWHEEL
> 14:26:23: ID_PANEL_PINWHEEL
> 14:26:23: ID_PANEL_PINWHEEL
> 14:26:23: ID_PANEL_PINWHEEL
> 14:26:23: ID_PANEL_PINWHEEL
> 14:26:23: ID_PANEL_RIPPLE
> 14:26:30: ID_PANEL_RIPPLE
> 14:26:31: ID_PANEL_RIPPLE
> 14:26:32: ID_PANEL_RIPPLE
> 14:26:33: ID_PANEL_RIPPLE
> 14:26:34: ID_PANEL_RIPPLE
> 14:26:35: ID_PANEL_RIPPLE
> 14:26:35: ID_PANEL_RIPPLE
> 14:26:36: ID_PANEL_RIPPLE
> 14:26:37: ID_PANEL_RIPPLE
> 14:26:38: ID_PANEL_RIPPLE
> 14:26:38: id
> 14:26:38: id
> 14:26:38: id
> 14:26:38: id
> 14:26:38: ID_PANEL_SHAPE
> 14:26:38: ID_PANEL_SHAPE
> 14:26:38: ID_PANEL_SHAPE
> 14:26:38: ID_PANEL_SHAPE
> 14:26:38: ID_PANEL_SHAPE
> 14:26:38: ID_PANEL_SHAPE
> 14:26:38: ID_PANEL_SHAPE
> 14:26:38: ID_PANEL_SHAPE
> 14:26:38: ID_PANEL_SHAPE
> 14:26:38: ID_PANEL_SHAPE
> 14:26:38: ID_PANEL_SHIMMER
> 14:26:38: ID_PANEL_SHIMMER
> 14:26:38: ID_PANEL_SHOCKWAVE
> 14:26:38: ID_PANEL_SHOCKWAVE
> 14:26:38: ID_PANEL_SHOCKWAVE
> 14:26:38: ID_PANEL_SHOCKWAVE
> 14:26:38: ID_PANEL_SHOCKWAVE
> 14:26:38: ID_PANEL_SHOCKWAVE
> 14:26:38: ID_PANEL_SINGLESTRAND
> 14:26:38: ID_PANEL_SINGLESTRAND
> 14:26:38: ID_PANEL_SINGLESTRAND
> 14:26:38: ID_PANEL_SINGLESTRAND
> 14:26:38: ID_PANEL_SINGLESTRAND
> 14:26:38: ID_PANEL_SINGLESTRAND
> 14:26:38: ID_PANEL_SKETCH
> 14:26:38: ID_PANEL_SKETCH
> 14:26:38: ID_PANEL_SKETCH
> 14:26:38: ID_PANEL_SNOWFLAKES
> 14:26:38: ID_PANEL_SNOWFLAKES
> 14:26:38: ID_PANEL_SPIRALS
> 14:26:38: ID_PANEL_SPIRALS
> 14:26:38: ID_PANEL_SPIRALS
> 14:26:38: ID_PANEL_SPIRALS
> 14:26:38: ID_PANEL_SPIROGRAPH
> 14:26:38: ID_PANEL_SPIROGRAPH
> 14:26:38: ID_PANEL_SPIROGRAPH
> 14:26:38: ID_PANEL_SPIROGRAPH
> 14:26:38: ID_PANEL_SPIROGRAPH
> 14:26:39: ID_PANEL_SPIROGRAPH
> 14:26:39: ID_PANEL_SPIROGRAPH
> 14:26:39: ID_PANEL_Tendril
> 14:26:39: ID_PANEL_Tendril
> 14:26:39: ID_PANEL_Tendril
> 14:26:39: ID_PANEL_Tendril
> 14:26:39: ID_PANEL_Tendril
> 14:26:39: ID_PANEL_Tendril
> 14:26:39: ID_PANEL_TWINKLE
> 14:26:39: ID_PANEL_TWINKLE
> 14:26:39: ID_PANEL_Video
> 14:26:39: ID_PANEL_Video
> 14:26:39: ID_PANEL_Video
> 14:26:39: ID_PANEL_Video
> 14:26:39: ID_PANEL_Video
> 14:26:39: ID_PANEL_VUMeter
> 14:26:39: ID_PANEL_VUMeter
> 14:26:39: ID_PANEL_WARP
> 14:26:39: ID_PANEL_WARP
> 14:26:39: ID_PANEL_WAVE
> 14:26:39: ID_PANEL_WAVE
> 14:26:39: ID_PANEL_WAVE
> 14:26:39: ID_PANEL_WAVE
> 14:26:39: ID_PANEL_WAVE
> 14:26:39: ID_PANEL_ON
> 14:26:39: ID_PANEL_COLORWASH
> 14:26:39: ID_PANEL_Music
> 14:26:39: ID_PANEL_PIANO
> 14:26:39: ID_PANEL_PLASMA
> 14:26:39: ID_PANEL_SERVO

**Comment by keithsw1111** [comment]
validatewindow is meant to be harmless when called a lot. While there may be opportunities to limit it in some circumstances it is not inherently a problem.
